libs/frontend/feature-build/src/lib/use-edit-component-styles.ts

```diff
@@ -0,0 +1,224 @@
+  editStyle: (prop: Css) => void
+  cancelEdit: (prop: Css) => void
+  setProperty: (oldEntry: IStyleEntry, newProp: Css) => void
+  setValue: (oldEntry: IStyleEntry, newValue: string) => void
+  saveStyle: (prop: Css) => void
+  removeStyle: (style: IStyleEntry) => void
+  addComponentStyle: () => void
+  getInheritedFrom: (entry: IInheritedStyleEntry) => string | undefined
+}
+
+export const useEditComponentStyles = (
+  options?: IUseEditComponenStylesOptions,
+): IUseEditComponentStyles => {
+  const styleType = options?.styleType
+  const { t } = useI18n()
+  const { site, currentPseudoClass, selectedComponentFlattenedStyles } = useBuild()
+
+  const styleEntries = computed(() =>
+    Object.entries(selectedComponentFlattenedStyles.value).map(
+      ([css, source]) =>
+        ({
+          pseudoClass: currentPseudoClass.value,
+          property: css as Css,
+          value: source.value,
+          sourceType: source.sourceType,
+          sourceId: source.sourceId,
+          sourceBreakpointId: source.sourceBreakpointId,
+        }) as IInheritedStyleEntry,
+    ),
+  )
+
+  const editStyle = (prop: Css) => {
+    setStyleType(styleType)
+    editStyles.value.add(prop)
+  }
+
+  const cancelEdit = (prop: Css) => {
+    editStyles.value.delete(prop)
+    checkEditStylesComplete()
+  }
+
+  // Finishes the editing session if no styles are active
+  const checkEditStylesComplete = () => {
+    if (editStyles.value.size === 0) {
+      editCommands.value = undefined
+    }
+  }
+
+  const setProperty = (oldEntry: IStyleEntry, newProp: Css) => {
+    const { pseudoClass, value } = oldEntry
+    editStyles.value.delete(oldEntry.property)
+    setStyle(
+      { pseudoClass, property: oldEntry.property, value },
+      { pseudoClass, property: newProp, value },
+    )
+    editStyles.value.add(newProp)
+  }
+
+  const setValue = (oldEntry: IStyleEntry, newValue: string) => {
+    const { pseudoClass, value, property } = oldEntry
+    setStyle({ pseudoClass, property, value }, { pseudoClass, property, value: newValue })
+  }
+
+  const removeEditCommand = (
+    prop: Css | undefined,
+  ): { removeCmd: ICommand | undefined; originalStyle: IStyleEntry | undefined } => {
+    const removeIndex = editCommands.value?.commands.findIndex((cmd) => {
+      const data = cmd.data as ISetComponentCustomStyleData
+      // Remove command if it resulted in `prop` being changed or removed
+      return data.newStyle?.property === prop || data.oldStyle?.property === prop
+    })
+    let removeCmd: ICommand<ISetComponentCustomStyleData> | undefined
+    if (removeIndex !== undefined && removeIndex !== -1) {
+      removeCmd = editCommands.value?.commands.splice(removeIndex, 1)[0] as
+        | ICommand<ISetComponentCustomStyleData>
+        | undefined
+    }
+    const originalStyle = removeCmd?.data.oldStyle
+      ? { ...removeCmd.data.oldStyle }
+      : undefined
+    return { removeCmd, originalStyle }
+  }
+
+  const setStyle = (
```

Comment:
This is pretty much the whole "preview" feature. It can probably be cleaned up a little more, but the idea is:
- When a style is updated, check for a command that affects the target property in the `editCommands` (used to track active style editing commands)
- If found, remove the command and replace it with the new one
- Replace the command in history, or add a new command on the first edit